crates/rattler_repodata_gateway/src/fetch/cache/mod.rs

```diff
@@ -80,6 +79,36 @@ impl FromStr for RepoDataState {
     }
 }
 
+/// Used inside of the `RepoDataState` to store information related to our JLAP state
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct JLAPState {
+    /// Initialization Vector (IV) for of the JLAP file; this is found on the first line of the
+    /// JLAP file.
+    pub iv: String,
+
+    /// Current position to use for the bytes offset in the range request for JLAP
+    pub pos: u64,
+
+    /// Footer contains metadata about the JLAP file such as which url it is for
+    pub footer: JLAPFooter,
+}
+
+/// Represents the metadata for a JLAP file, which is typically found at the very end
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct JLAPFooter {
+    /// URL of the `repodata.json` file
+    pub url: Option<String>,
+
+    /// blake2b hash of the latest `repodata.json` file
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_blake2_hash",
+        serialize_with = "serialize_blake2_hash"
+    )]
+    pub latest: Option<blake2::digest::Output<Blake2b256>>,
```

Comment:
Technically, yes. I did this so that `JLAPState::default()` would work, so we could generate an "empty" JLAPState. When a JLAP request fails, for example when the range request fails or it can't find the correct starting point for patches to apply, we need a way to reset the cached JLAP values.

Now that I think about it though, I think we could actually just set the `jlap` value in cache to None instead. That would be a better more explicit (i.e. we are purposely store nothing here versus a default `JLAPState`).

Reply:
This is where I could make those changes:

https://github.com/travishathaway/rattler/blob/6e8669cbd35a327cb1781f56f6cf4dd7379a5dad/crates/rattler_repodata_gateway/src/fetch/mod.rs#L376-L380